### DISCUSSION TITLE
How can I add a custom helper method in a model?

### DISCUSSION THREAD

**Discussion by vsahinidroot** [prisma/prisma#16681] [Q&A]
Hello!  How can I add a custom helper method in a model using Express?  For example what if I want to have custom pre and post-save hooks and additional custom methods such as `generateJWTToken(self)` where `self` is the instance of the model like other ORMs provide? I can't add these Javascript methods in the Prisma schema file so what would be best-practice?

Thanks!

**Comment by nikolasburk** [comment]
Hey @vsahinidroot 👋 you can use the new Prisma Client extensions feature for exactly that, here's the [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/model) that show how you can add custom methods to your models 🙌

**Reply by vladi-stevanovic** [reply]
👋 @vsahinidroot, did you perhaps have a chance to check Niko's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
